packages/core-components/src/layout/SignInPage/guestProvider.tsx

```diff
@@ -20,39 +20,55 @@ import Button from '@material-ui/core/Button';
 import { InfoCard } from '../InfoCard/InfoCard';
 import { GridItem } from './styles';
 import { ProviderComponent, ProviderLoader, SignInProvider } from './types';
-import { GuestUserIdentity } from './GuestUserIdentity';
-
-const Component: ProviderComponent = ({ onSignInStarted, onSignInSuccess }) => (
-  <GridItem>
-    <InfoCard
-      title="Guest"
-      variant="fullHeight"
-      actions={
-        <Button
-          color="primary"
-          variant="outlined"
-          onClick={() => {
-            onSignInStarted();
-            onSignInSuccess(new GuestUserIdentity());
-          }}
-        >
-          Enter
-        </Button>
-      }
-    >
-      <Typography variant="body1">
-        Enter as a Guest User.
-        <br />
-        You will not have a verified identity,
-        <br />
-        meaning some features might be unavailable.
-      </Typography>
-    </InfoCard>
-  </GridItem>
-);
-
-const loader: ProviderLoader = async () => {
-  return new GuestUserIdentity();
+import { ProxiedSignInIdentity } from '../ProxiedSignInPage/ProxiedSignInIdentity';
+import { discoveryApiRef, useApi } from '@backstage/core-plugin-api';
+
+const Component: ProviderComponent = ({ onSignInStarted, onSignInSuccess }) => {
+  const discoveryApi = useApi(discoveryApiRef);
+  return (
+    <GridItem>
+      <InfoCard
+        title="Guest"
+        variant="fullHeight"
+        actions={
+          <Button
+            color="primary"
+            variant="outlined"
+            onClick={() => {
+              onSignInStarted();
+              onSignInSuccess(
+                new ProxiedSignInIdentity({
+                  provider: 'guest',
+                  discoveryApi,
+                }),
+              );
+            }}
+          >
+            Enter
+          </Button>
+        }
+      >
+        <Typography variant="body1">Sign in as a Guest.</Typography>
+      </InfoCard>
+    </GridItem>
+  );
+};
+
+const loader: ProviderLoader = async apis => {
+  const identity = new ProxiedSignInIdentity({
+    provider: 'guest',
+    discoveryApi: apis.get(discoveryApiRef)!,
+  });
+
+  await identity.start();
+
+  const identityResponse = await identity.getBackstageIdentity();
+
+  if (!identityResponse) {
+    return undefined;
+  }
+
+  return identity;
```

Comment:
Do we need these lines at all? Spontaneously it might be ok to only call start, or to get an identity, not both?

Reply:
Updated, see above comment.